.github/workflows/npm-publish.yaml

```diff
@@ -0,0 +1,53 @@
+name: Publish to npm
+
+on:
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18.18.2'
+          registry-url: 'https://registry.npmjs.org'
+
+        name: Install dependencies    
+      - run: yarn install --frozen-lockfile
+
+        name: Run ESLint
+      - run: yarn lint
+
+        name: Check TypeScript errors
+      - run: yarn type-check
+
+  publish:
+    needs: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18.18.2'
+          registry-url: https://registry.npmjs.org/
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Update package version
+        run: |
+          # Get current version from package.json
+          CURRENT_VERSION=$(node -p "require('./package.json').version")
+          # Create a new version with timestamp (Format: 0.1.0-202328111704.rc0)
+          TIMESTAMP=$(date +"%Y%m%d%H%M")
+          NEW_VERSION="${CURRENT_VERSION}-${TIMESTAMP}.rc0"
+          echo "New version: $NEW_VERSION"
+          # Update the version in package.json
+          npm version $NEW_VERSION --no-git-tag-version
```

Comment:
I think you can change it to just `npm version prerelease --preid="$(date +'%y%m%d%H%M%S')"`. `rc` increments and will be added automatically

Reply:
awesome, it works, thank you for the short solution